src/js/player.js

```diff
@@ -823,6 +825,112 @@ class Player extends Component {
   }
 
   /**
+   * Modify the video/audio element so that we can detect when
+   * the source is changed. Fires `sourceset` just after the source has changed
+   *
+   * @fires Player#sourceset
+   */
+  watchForSourceSet_() {
+    if (browser.IS_IE8) {
+      return;
+    }
+    // if we cannot overwrite the src property, there is no support
+    // iOS 7 safari for instance cannot do this.
+    try {
+      Object.defineProperty(document.createElement('video'), 'src', {get: () => {}, set: () =>{}});
+    } catch (e) {
+      return;
+    }
+
+    const el = this.$('video') || this.$('audio');
+
+    if (!el) {
+      if (!this.isReady_) {
+        this.ready(() => this.watchForSourceSet_());
+      }
+      return;
+    }
+
+    // we need to fire sourceset when the player is ready
+    // if we find that the media element had a src when it was
+    // given to us
+    if (this.tagAttributes && this.tagAttributes.src) {
+      this.ready(() => this.trigger('sourceset'));
+    }
+
+    const proto = window.HTMLMediaElement.prototype;
+    let srcDescriptor = {};
+
+    // preserve getters/setters already on `el.src` if they exist
+    if (Object.getOwnPropertyDescriptor(el, 'src')) {
+      srcDescriptor = Object.getOwnPropertyDescriptor(el, 'src');
+    } else if (Object.getOwnPropertyDescriptor(proto, 'src')) {
+      srcDescriptor = mergeOptions(srcDescriptor, Object.getOwnPropertyDescriptor(proto, 'src'));
+    }
+
+    if (!srcDescriptor.get) {
+      srcDescriptor.get = function() {
+        return proto.getAttribute.call(this, 'src');
+      };
+    }
+
+    if (!srcDescriptor.set) {
+      srcDescriptor.set = function(v) {
+        return proto.setAttribute.call(this, 'src', v);
+      };
+    }
+
+    if (typeof srcDescriptor.enumerable === 'undefined') {
+      srcDescriptor.enumerable = true;
+    }
+
+    Object.defineProperty(el, 'src', {
+      get: srcDescriptor.get.bind(el),
+      set: (v) => {
+        const retval = srcDescriptor.set.call(el, v);
+
+        this.ready(() => this.trigger('sourceset'), true);
+
+        return retval;
+      },
+      configurable: true,
+      enumerable: srcDescriptor.enumerable
+    });
+
+    const oldSetAttribute = el.setAttribute;
+
+    el.setAttribute = (n, v) => {
```

Comment:
does setting the `src` via `setAttribute` actually change the source of the element? if not, we could then skip this.

Reply:
setAttribute does actually change the source